File: ICPC/INSATCPC2018/gen.py
from random import randint

N = 100000
Q = 100000
MAXN = 100000
print(N, Q)
a = ' '.join([str(randint(1,MAXN)) for i in range(N)])
print(a)
for i in range(Q):
	q = []
	q.append(str(randint(1, 2)))
	l = randint(1, N)
	d = randint(0, N - l)
	q.append(str(l))
	q.append(str(l + d))
	print(' '.join(q))

File: CodeChef/gen.cpp
#include "../jngen.h"

#include <iostream>
#include <sstream>
#include <fstream>
#include <iomanip>
#include <string>
#include <cstdlib>
#include <cstdio>
#include <cstring>
#include <cmath>
#include <ctime>
#include <climits>
#include <cassert>
#include <vector>
#include <queue>
#include <stack>
#include <deque>
#include <set>
#include <map>
#include <bitset>
#include <utility>
#include <algorithm>

#define forn(i, n) for (int i = 0; i < int(n); i++)

using namespace std;

const int MAXN = 1000;

int main(int argc, char* argv[])
{
    srand(time(NULL));
    parseArgs(argc, argv);
    int n;
    n = rand() % MAXN + 1;
    setMod().add1();
    cout << 1 << '\n';
    cout << n << " " << '\n';
    Tree t = Tree::random(n);
    t.setEdgeWeights(Array::random(n - 1,100000));
    cout << t << endl;
}
